core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -858,13 +893,8 @@ private[zio] final class FiberContext[E, A](
   ): Unit = {
     val result = Exit.succeed(v)
 
-    // To preserve fair scheduling, we submit all resumptions on the thread
-    // pool in order of their submission.
-    observers.reverse.foreach(
-      k =>
-        platform.executor
-          .submitOrThrow(() => k(result))
-    )
+    // For improved fairness, we resume in order of submission:
+    observers.reverse.foreach(k => k(result))
```

Comment:
Hmm why did we stop resuming on the threadpool here? Does this not execute the resumptions sequentially?

Reply:
It does, but now async operations just "trigger" resumption of the computation on the normal ZIO thread pool, so this improves performance because it eliminates the extra unnecessary redundant submission.